app/src/main/java/com/infomaniak/mail/ui/main/move/MoveAdapter.kt

```diff
@@ -25,175 +25,81 @@ import androidx.recyclerview.widget.DiffUtil
-import com.infomaniak.mail.views.itemViews.*
+import com.infomaniak.mail.views.itemViews.SelectableFolderItemView
+import com.infomaniak.mail.views.itemViews.SelectableItemView
 import javax.inject.Inject
 import kotlin.math.min
 
 class MoveAdapter @Inject constructor() : ListAdapter<Folder, FolderViewHolder>(FolderDiffCallback()) {
 
     private var sourceFolderId: String? = null
-    private var hasCollapsableFolder: Boolean? = null
-    private var isInMenuDrawer: Boolean = true
 
     private lateinit var onFolderClicked: (folderId: String) -> Unit
     private var onCollapseClicked: ((folderId: String, shouldCollapse: Boolean) -> Unit)? = null
-    private var onCollapseTransition: ((Boolean) -> Unit)? = null
 
     operator fun invoke(
-        isInMenuDrawer: Boolean,
         onFolderClicked: (folderId: String) -> Unit,
         onCollapseClicked: ((folderId: String, shouldCollapse: Boolean) -> Unit)? = null,
-        onCollapseTransition: ((Boolean) -> Unit)? = null,
     ): MoveAdapter {
-        this.isInMenuDrawer = isInMenuDrawer
         this.onFolderClicked = onFolderClicked
         this.onCollapseClicked = onCollapseClicked
-        this.onCollapseTransition = onCollapseTransition
 
         return this
     }
 
-    fun setSourceFolderId(id: String) {
-        sourceFolderId = id
+    fun setFolders(newSourceFolderId: String, newFolders: List<Folder>) = runCatchingRealm {
+        sourceFolderId = newSourceFolderId
+        submitList(newFolders)
     }
 
     override fun getItemCount(): Int = runCatchingRealm { currentList.size }.getOrDefault(0)
 
-    override fun getItemViewType(position: Int): Int {
-        return if (isInMenuDrawer) DisplayType.MENU_DRAWER.layout else DisplayType.SELECTABLE_FOLDER.layout
-    }
+    override fun getItemViewType(position: Int): Int = R.layout.item_selectable_folder
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FolderViewHolder {
         val layoutInflater = LayoutInflater.from(parent.context)
-        val binding = if (viewType == DisplayType.SELECTABLE_FOLDER.layout) {
-            ItemSelectableFolderBinding.inflate(layoutInflater, parent, false)
-        } else {
-            ItemMenuDrawerFolderBinding.inflate(layoutInflater, parent, false)
-        }
-
+        val binding = ItemSelectableFolderBinding.inflate(layoutInflater, parent, false)
         return FolderViewHolder(binding)
     }
 
     override fun onBindViewHolder(holder: FolderViewHolder, position: Int, payloads: MutableList<Any>) = runCatchingRealm {
         if (payloads.firstOrNull() == Unit) {
-            val folder = currentList[position]
-            if (getItemViewType(position) == DisplayType.SELECTABLE_FOLDER.layout) {
-                (holder.binding as ItemSelectableFolderBinding).root.setSelectedState(sourceFolderId == folder.id)
-            }
+            (holder.binding as ItemSelectableFolderBinding).root.setSelectedState(sourceFolderId == currentList[position].id)
```

Comment:
If there's only one view type, we could directly use the correct view binding type so we don't need to cast it

Reply:
True.
But there will bill 2 viewTypes in the next PR, so let's just ignore it for the moment.